lib/infomon/activespell.rb

```diff
@@ -0,0 +1,107 @@
+module ActiveSpell
+  #
+  # Spell timing true-up (Invoker and SK item spells do not have proper durations)
+  # this needs to be addressed in class Spell rewrite
+  # in the meantime, this should mean no spell is more than 1 second off from
+  # Simu's time calculations
+  #
+
+  def self.get_spell_info
+    puts "spell update requested\r\n" if $infomon_debug
+    @update_spell_durations = XMLData.active_spells
+    @update_spell_names = []
+    @makeychange = []
+    @update_spell_durations.each do |k, _v|
+      case k
+      when /(?:Mage Armor|520) - /
+        @makeychange << k
+        @update_spell_names.push('Mage Armor')
+        next
+      when /(?:CoS|712) - /
+        @makeychange << k
+        @update_spell_names.push('Cloak of Shadows')
+        next
+      when /Enh\./
+        @makeychange << k
+        case k
+        when /Enh\. Strength/
+          @update_spell_names.push('Surge of Strength')
+        when /Enh\. (?:Dexterity|Agility)/
+          @update_spell_names.push('Burst of Swiftness')
+        end
+        next
+      when /Empowered/
+        @makeychange << k
+        @update_spell_names.push('Shout')
+        next
+      when /Multi-Strike/
+        @makeychange << k
+        @update_spell_names.push('MStrike Cooldown')
+        next
+      when /Next Bounty Cooldown/
+        @makeychange << k
+        @update_spell_names.push('Next Bounty')
+        next
+      end
+      @update_spell_names << k
+    end
+    @makeychange.each do |changekey|
+      next unless @update_spell_durations.key?(changekey)
+
+      case changekey
+      when /(?:Mage Armor|520) - /
+        @update_spell_durations['Mage Armor'] = @update_spell_durations.delete changekey
+      when /(?:CoS|712) - /
+        @update_spell_durations['Cloak of Shadows'] = @update_spell_durations.delete changekey
+      when /Enh\. Strength/
+        @update_spell_durations['Surge of Strength'] = @update_spell_durations.delete changekey
+      when /Enh\. (?:Dexterity|Agility)/
+        @update_spell_durations['Burst of Swiftness'] = @update_spell_durations.delete changekey
+      when /Empowered/
+        @update_spell_durations['Shout'] = @update_spell_durations.delete changekey
+      when /Multi-Strike/
+        @update_spell_durations['MStrike Cooldown'] = @update_spell_durations.delete changekey
+      when /Next Bounty Cooldown/
+        @update_spell_durations['Next Bounty'] = @update_spell_durations.delete changekey
+      when /Next Group Bounty Cooldown/
+        @update_spell_durations['Next Group Bounty'] = @update_spell_durations.delete changekey
+      end
+    end
+  end
+
+  Thread.new do
+    loop do
+      begin
+        sleep 0.01 until XMLData.process_spell_durations
+        ActiveSpell.get_spell_info
```

Comment:
this should look like `[update_spell_durations, update_spell_names] = ActiveSpell.get_spell_info`